private/buf/bufctl/controller.go

```diff
@@ -1287,6 +1287,15 @@ func (c *controller) handleFileAnnotationSetRetError(retErrAddr *error) {
 	}
 }
 
+// Implements [protoyaml.Validator] using [protovalidate.Validator]. This allows us to pass
+// a [protovalidate.Validator] to the [protoencoding.NewYAMLUnmarshaler] and validate types
+// when unmarshalling.
+type yamlValidator struct{ protovalidateValidator protovalidate.Validator }
```

Comment:
This type doesn't look necessary as the interface is the same: https://pkg.go.dev/buf.build/go/protoyaml#Validator and https://pkg.go.dev/github.com/bufbuild/protovalidate-go#Validator .

Reply:
The current version of `Validator.Validate` takes variadic `ValidationOption`s: https://github.com/bufbuild/protovalidate-go/blob/38a17488914d10dba0b143496b8d43541e6e3378/validator.go#L33-L41